airgun/browser.py

```diff
@@ -298,6 +261,58 @@ def _get_docker_browser(self):
         self._webdriver = self._docker.webdriver
         return self._webdriver
 
+    def _get_remote_browser(self):
+        """Returns remote webdriver instance of selected ``browser``.
+
+        Note: should not be called directly, use :meth:`get_browser` instead.
+        """
+        self._webdriver = webdriver.Remote(
+            command_executor=settings.selenium.command_executor,
+            desired_capabilities=self._get_webdriver_capabilities()
+        )
+        idle_timeout = settings.webdriver_desired_capabilities.idleTimeout
+        if idle_timeout:
+            self._webdriver.command_executor.set_timeout(int(idle_timeout))
+        # todo: attempt to rename job here
```

Comment:
This comment seems to be copied from `_get_saucelab_browser`. Does it make sense in context of remote browser as well?

Reply:
Yes, `_get_remote_browser` has the same requirement for webdriver_capabilities, as you said, saucelab is also a remote mode, the only difference is how to define command_executor.

Reply:
I mean this comment specifically:

> todo: attempt to rename job here

SauceLabs has test runner and test reporter capabilities, so this comment kind of makes sense to me in that context - we would like to change "job name", to have something better in SauceLabs (but why and to what exactly escapes me).

But in context of remote webdriver, I don't understand what would be a "job", why we would like to rename it and what name we would like to use instead.
Perhaps this is just me missing something about remote webdriver, but if you could explain it, I would very much appreciate it.

Reply:
@mirzal saucelabs doesn't know test name, so when session starts, it just gives it unique id. After test finishes, we send a request to update session name to test name and result to either "pass" or "fail". This isn't great by 2 reasons:
1) while tests are running (during test run or when you're debugging smth) its hard to find proper session as its one of those 8+ "d7f9sd97fsd9fsd7fs9d7f" running in parralel
2) if, for some reason, connection with saucelabs breaks, its pretty much impossible to find out what test caused issues using saucelabs interface only (of course, possible via jenkins logs) due to that id instead of name.

So i left this comment to return back in future and attempt to rename job immideately after session start, as it could be a nice improvement (but nothing really critical, just handy in some rare cases).

And this comment is irrelevant for remote webdriver for sure.

Reply:
Okay, I copied the code from sauce because the same structure in my understanding, I will remove it for remote provider, thank you all.